src/features/defi/providers/univ2/components/UniV2Manager/Deposit/components/Deposit.tsx

```diff
@@ -120,63 +127,63 @@ export const Deposit: React.FC<DepositProps> = ({
         title: translate('common.somethingWentWrong'),
         status: 'error',
       })
-      dispatch({ type: FoxEthLpDepositActionType.SET_LOADING, payload: false })
+      dispatch({ type: UniV2DepositActionType.SET_LOADING, payload: false })
     }
   }
 
   const handleContinue = async (formValues: DepositValues) => {
-    if (!foxEthLpOpportunity) return
+    if (!lpOpportunity) return
     // set deposit state for future use
     dispatch({
-      type: FoxEthLpDepositActionType.SET_DEPOSIT,
+      type: UniV2DepositActionType.SET_DEPOSIT,
       payload: {
-        ethCryptoAmount: formValues.cryptoAmount0,
-        ethFiatAmount: formValues.fiatAmount0,
-        foxCryptoAmount: formValues.cryptoAmount1,
-        foxFiatAmount: formValues.fiatAmount1,
+        asset0CryptoAmount: formValues.cryptoAmount0,
+        asset0FiatAmount: formValues.fiatAmount0,
+        asset1CryptoAmount: formValues.cryptoAmount1,
+        asset1FiatAmount: formValues.fiatAmount1,
       },
     })
     trackOpportunityEvent(
       MixPanelEvents.DepositContinue,
       {
-        opportunity: foxEthLpOpportunity,
+        opportunity: lpOpportunity,
         fiatAmounts: [formValues.fiatAmount0, formValues.fiatAmount1],
         cryptoAmounts: [
-          { assetId: ethAssetId, amountCryptoHuman: formValues.cryptoAmount0 },
-          { assetId: foxAssetId, amountCryptoHuman: formValues.cryptoAmount1 },
+          { assetId: assetId0, amountCryptoHuman: formValues.cryptoAmount0 },
+          { assetId: assetId1, amountCryptoHuman: formValues.cryptoAmount1 },
         ],
       },
       assets,
     )
-    dispatch({ type: FoxEthLpDepositActionType.SET_LOADING, payload: true })
+    dispatch({ type: UniV2DepositActionType.SET_LOADING, payload: true })
     try {
       // Check if approval is required for user address
       const lpAllowance = await allowance()
-      const allowanceAmount = bnOrZero(lpAllowance).div(bn(10).pow(foxAsset.precision))
+      const allowanceAmount = bnOrZero(lpAllowance).div(bn(10).pow(asset1.precision))
```

Comment:
might need to generalize this to handle the case where ETH doesn't need an approval